Vagrantfile

```diff
@@ -0,0 +1,43 @@
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "hashicorp/bionic64"
+  
+  config.vm.provision "shell", privileged: false, inline: <<-SHELL 
+    sudo apt-get update
+    
+    sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+      libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+      xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.bashrc
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bashrc
+    echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.bashrc
+    
+    # sourcing ~/.bashrc doesn't seem to work...
```

Comment:
Not sure why... any thoughts?

Reply:
First thing you should fix:
Your line below - ` echo 'eval "$(pyenv init -)"' > ~/.bashrc` - is overriding your previous additions to .bashrc and results in a broken .bashrc file. It should be deleted

If you fix that, and then ssh into your vagrant box, you can look at ~/.bashrc. It's a long file but at the very top is an "if not in an interactive shell, don't do anything". This setup is being run in an uninteractive shell.

So use .profile rather than .bashrc

And you want .profile rather than .bashrc for setup like env vars anyway (one time setup rather than something that also gets run again for subshells)